source/slang/slang-ir-any-value-inference.cpp

```diff
@@ -0,0 +1,231 @@
+#include "slang-ir-any-value-inference.h"
+
+#include "slang-ir-generics-lowering-context.h"
+#include "slang-ir.h"
+#include "slang-ir-insts.h"
+#include "slang-ir-util.h"
+#include "slang-ir-layout.h"
+#include "../core/slang-func-ptr.h"
+
+namespace Slang
+{
+
+    void _findDependenciesOfTypeInSet(IRType* type, HashSet<IRInterfaceType*>& targetSet, List<IRInterfaceType*>& result)
+    {
+        switch (type->getOp())
+        {
+        case kIROp_InterfaceType:
+            {
+                auto interfaceType = cast<IRInterfaceType>(type);
+                if (targetSet.contains(interfaceType))
+                {
+                    result.add(interfaceType);
+                    return;
+                }
+            }
+            break;
+        case kIROp_StructType:
+            {
+                auto structType = cast<IRStructType>(type);
+                for (auto field : structType->getFields())
+                {
+                    _findDependenciesOfTypeInSet(field->getFieldType(), targetSet, result);
```

Comment:
While recursion is convenient, we should try to use worklists whenever possible to avoid stack related issues.